pages/dashboard/index.js

```diff
@@ -1,10 +1,85 @@
+const moment = require('moment');
+const { cloneDeep } = require('lodash');
 const { page } = require('@asl/service/ui');
+const { dateFormat } = require('@asl/pages/constants');
 const taskList = require('@asl/pages/pages/task/list/router');
 
-function userIsNtcoAtEst(profile, estId) {
-  return !!profile.roles.find(role => role.establishmentId === estId && role.type === 'ntco');
+function getAlertUrl(alert, buildRoute) {
+  switch (alert.type) {
+    case 'pilReview':
+      return buildRoute('pil.read', {
+        establishmentId: alert.model.establishmentId,
+        profileId: alert.model.profileId,
+        pilId: alert.model.id
+      });
+
+    case 'raDue':
+      return buildRoute('project.read', {
+        establishmentId: alert.model.establishmentId,
+        projectId: alert.model.id
+      });
+
+    case 'ropDue':
+      return buildRoute('project.read', {
+        establishmentId: alert.model.establishmentId,
+        projectId: alert.model.id
+      });
+  }
 }
 
+/**
+ * Group alerts by establishment, type and status e.g.
+ * [
+ *   {
+ *     id,
+ *     name,
+ *     summary: {
+ *       pilReview: { due: 0, overdue: 0 },
+ *       raDue: { due: 0, overdue: 0 },
+ *       ropDue: { due: 0, overdue: 0 }
+ *     }
+ *   },
+ *   ...
+ * ]
+ */
+const summariseEstablishmentAlerts = (alerts = [], profileEstablishments = [], buildRoute) => {
+  const summary = {
+    pilReview: { due: 0, overdue: 0 },
+    raDue: { due: 0, overdue: 0 },
+    ropDue: { due: 0, overdue: 0 }
+  };
+
+  return alerts.reduce((result, alert) => {
+    let idx = result.findIndex(e => e.id === alert.establishmentId);
+
+    const establishment = idx !== -1
+      ? result[idx]
+      : {
+        id: alert.establishmentId,
+        name: (profileEstablishments.find(e => e.id === alert.establishmentId) || {}).name,
+        summary: cloneDeep(summary)
+      };
+
+    if (alert.overdue) {
+      establishment.summary[alert.type].overdue++;
+    } else {
+      establishment.summary[alert.type].due++;
+    }
+
+    if (idx !== -1) {
+      result[idx] = establishment;
+    } else {
+      establishment.summary['pilReview'].url = buildRoute('pils', { establishmentId: alert.establishmentId });
+      const raUrlParams = '?status=inactive-statuses&sort%5Bcolumn%5D=raDate&sort%5Bascending%5D=true&page=1';
+      establishment.summary['raDue'].url = buildRoute('project', { establishmentId: alert.establishmentId, suffix: raUrlParams });
+      establishment.summary['ropDue'].url = buildRoute('establishment.rops', { establishmentId: alert.establishmentId });
+      result.push(establishment);
+    }
+
+    return result;
+  }, []);
+};
```

Comment:
This is probably a bit cleaner if you reduce into a keyed object then return the `Object.values()` rather than having to mess with array indexes.

```js
const establishments = alerts.reduce((result, alert) => {

  const raQueryString = '?status=inactive-statuses&sort%5Bcolumn%5D=raDate&sort%5Bascending%5D=true&page=1';
  const urlParams = { establishmentId: alert.establishmentId };
  const establishment = result[alert.establishmentId] || {
    id: alert.establishmentId,
    name: (profileEstablishments.find(e => e.id === alert.establishmentId) || {}).name,
    summary: {
      pilReview: {
        due: 0,
        overdue: 0,
        url = buildRoute('pils', urlParams)
      },
      raDue: {
        due: 0,
        overdue: 0,
        url = buildRoute('project', { ...urlParams, suffix: raQueryString })
      },
      ropDue: {
        due: 0,
        overdue: 0,
        buildRoute('establishment.rops', urlParams)
      }
    }
  };

  if (alert.overdue) {
    establishment.summary[alert.type].overdue++;
  } else {
    establishment.summary[alert.type].due++;
  }

  return { ...result, [establishment.id]: establishment };
}, {});

return Object.values(establishments);
```